question/bank/previewquestion/classes/previewquestion_helper.php

```diff
@@ -249,11 +261,106 @@ public static function get_preview_extra_elements(\question_definition $question
         foreach ($plugins as $componentname => $plugin) {
             $pluginhtml = component_callback($componentname, $functionname, [$question, $courseid]);
             if ($componentname === 'qbank_comment') {
-                $comment = $pluginhtml;
+                $comment = self::preview_comment($question->id, $context);
                 continue;
             }
             $extrahtml[] = $pluginhtml;
         }
         return [$comment, $extrahtml];
     }
+
+    /**
+     * Checks if question is the latest version.
+     *
+     * @param string $version Question version to check.
+     * @param string $questionbankentryid Entry to check against.
+     * @return bool
+     */
+    public static function is_latest(string $version, string $questionbankentryid) : bool {
+        global $DB;
+
+        $sql = 'SELECT MAX(version)
+                  FROM {question_versions}
+                 WHERE questionbankentryid = ?';
+        $latestversion = $DB->get_record_sql($sql, [$questionbankentryid]);
+
+        return ($version === $latestversion->max) ? true : false;
+    }
+
+    /**
+     * Renders question preview cog wheel menu.
+     *
+     * @param  object $question Question informations.
+     * @return string $menu Cog wheel menu to render.
+     */
+    public function display_edit_menu(object $question) : string {
```

Comment:
i will have some comments here, but I cant test in locally as I am getting an error, please fix that error then I can see this part.

Reply:
Error fixed